renku/core/models/cwl/command_line_tool.py

```diff
@@ -271,6 +271,8 @@ def generate_tool(self):
     @contextmanager
     def watch(self, client, no_output=False):
         """Watch a Renku repository for changes to detect outputs."""
+        client.check_external_storage()
```

Comment:
in a similar vein to the one mentioned above, we could decorate `with_workflow_storage` with `@ensure_workflow_storage` to achieve the same result, then `check_external_storage` is not needed

Reply:
Same as above.